### PR TITLE
feat: List Notifications API

### PR DESCRIPTION
**Pull request by manishdangi98**
Add List Notification API in Siren.
This API list notifications based on the filter passed.
Supported filters are Type, Template, ReceiverSelectors and Labels.

### CODE REVIEW

File: internal/api/v1beta1/notification.go [hunk 1]
```diff
@@ -94,3 +97,42 @@ func (s *GRPCServer) ListNotificationMessages(ctx context.Context, req *sirenv1b
...
+	items := []*sirenv1beta1.Notification{}
+
+	for _, notification := range notifications {
+		data, err := structpb.NewStruct(notification.Data)
+		if err != nil {
+			return nil, s.generateRPCErr(err)
+		}
+
+		currentTime := time.Now()
+		validDuration := currentTime.Add(notification.ValidDuration)
```
Comment (mabdh) [inline internal/api/v1beta1/notification.go:121]: why do we need to do this?

File: internal/store/postgres/notification_test.go [hunk 2]
```diff
@@ -118,6 +128,145 @@ func (s *NotificationRepositoryTestSuite) TestCreate() {
...
+		{
+			Description: "should get all notification with type filter",
+			Filter: notification.Filter{
+				Type: "receiver",
+			},
+			ExpectedNotifications: []notification.Notification{
+				{
+					ID:          "123",
+					NamespaceID: 1,
+					Type:        "receiver",
```
Comment (mabdh) [inline internal/store/postgres/notification_test.go:180]: Let's update this to either `alert` or `event`.
Let's update the mock files too

File: internal/store/postgres/notification_test.go [hunk 3]
```diff
@@ -118,6 +128,145 @@ func (s *NotificationRepositoryTestSuite) TestCreate() {
...
+					},
+					Labels:        map[string]string{},
+					ValidDuration: time.Duration(0),
+					Template:      "alert",
+					UniqueKey:     "",
+				},
+			},
+		},
+		{
+			Description: "should get all notification with lable filter",
```
Comment (mabdh) [inline internal/store/postgres/notification_test.go:212]:
```suggestion
			Description: "should get all notifications with labels filter",
```

File: internal/store/postgres/notification_test.go [hunk 4]
```diff
@@ -118,6 +128,145 @@ func (s *NotificationRepositoryTestSuite) TestCreate() {
...
+	type testCase struct {
+		Description           string
+		Filter                notification.Filter
+		ExpectedNotifications []notification.Notification
+		ErrString             string
+	}
+
+	var testCases = []testCase{
+		{
+			Description: "should get all notification with receiver_selector filter",
```
Comment (mabdh) [inline internal/store/postgres/notification_test.go:141]:
```suggestion
			Description: "should get all notifications with receiver_selector filter",
```

File: internal/api/v1beta1/notification.go (hunk 1 above)
Comment (manishdangi98) [reply internal/api/v1beta1/notification.go:121]: In the proto ValidDuration is expecting `*timestamppb.Timestamp`  but we are getting `time.Duration`. 
That is reason for conversion but  need to make change in proto and check if it supports `time.Duration` type.

Comment (mabdh) [reply internal/api/v1beta1/notification.go:121]: Then the proto is wrong right? This valid duration is not a timestamp

Comment (manishdangi98) [reply internal/api/v1beta1/notification.go:121]: Yes @mabdh

**Review by mabdh — APPROVED** [review]

**Pull request merged**